### PR TITLE
Add IP address markers

### LINKED ISSUE #163: Add IP address markers
## Overview

Implements the `govalid:ipv4` `govalid:ipv6` `govalid:ip` markers for validating IP addresses.

## Usage

```go
type SampleRequest struct {
    // +govalid:ipv4
    ClientIPv4 string

    // +govalid:ipv6
    ClientIPv6 string

    // +govalid:ip
    ClientIP string
}
```

## Expected Behavior

`govalid:ipv4`

Validate that the format complies with RFC 791 for IPv4
e.g. 1.1.1.1

`govalid:ipv6`

Validate that the format complies with RFC 4291 for IPv6
e.g. 2001:0db8:85a3:0000:0000:8a2e:0370:7334

`govalid:ip`

Validate that the format complies with either IPv4 or IPv6 standards
e.g. 1.1.1.1 or 2001:0db8:85a3:0000:0000:8a2e:0370:7334

## To the maintainer

If it's convenient, I'd appreciate it if you could assign this issue to me.

One point of consideration is whether it's better to split the marker into three separate entries as shown above.
I was also considering alternatives like `govalid:ip=v4`, `govalid:ip=v6`, and `govalid:ip=all`.

Please advise which approach would be preferable.

### PR DESCRIPTION
**Pull request by t4kamura**
Hello, I added IP validation so please review it when you have time 😄 

## Description

#163
Implements the `govalid:ipv4` `govalid:ipv6` markers for validating IP addresses.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)

## Quality Checklist

Before submitting this pull request, ensure your contribution meets these quality standards:

### Code Quality
- [x] Code passes `make golangci-lint` without errors
- [x] All tests pass: `make test`
- [x] Code follows existing patterns and conventions
- [x] No hard-coded values (use constants or configuration)
- [x] Error messages are clear and actionable

### Testing Requirements
- [x] Golden tests created and passing (`internal/analyzers/govalid/testdata/`)
- [x] Unit tests implemented with boundary value testing (`test/unit/`)
- [x] Benchmark tests comparing against popular validation libraries (`test/benchmark/`)
  - [x] [go-playground/validator](https://github.com/go-playground/validator) (BenchmarkGoPlayground*)
  - [ ] [asaskevich/govalidator](https://github.com/asaskevich/govalidator) (BenchmarkGoValidator*)
  - [ ] [gookit/validate](https://github.com/gookit/validate) (BenchmarkGookitValidate*)
- [x] All test files generated: `cd test && go generate`

### Performance Standards
- [x] Zero allocations in validation logic (verified via benchmarks)
- [x] Performance improvement over existing validation libraries (minimum 2x faster)
- [ ] Benchmark results added to `test/benchmark/README.md`

### Documentation
- [x] README.md updated with new marker documentation
- [x] Code includes appropriate comments and examples
- [x] Validator usage examples provided

### Integration
- [x] Validator scaffold generated with `make generate-validator MARKER=yourmarker`
- [x] Registry files automatically updated
- [x] Binary rebuilt and installed: `go install ./cmd/govalid/`

### Final Verification
- [ ] All GitHub Actions CI checks pass
- [x] No breaking changes to existing functionality
- [x] Contribution follows project license requirements

## Additional Notes

Used the `net` package for IP validation.
Noteworthy point: variables are defined within the if scope.

```go
return fmt.Sprintf("ip := net.ParseIP(t.%s); ip == nil || ip.To4() == nil", v.FieldName())
```

https://github.com/t4kamura/govalid/blob/28e38b25a5a460054cdf7474e28c9c3d2e234e26/internal/validator/rules/ipv4.go?plain=1#L28

I debated whether to write this as a helper function, but couldn't fit it on a single line, so I defined the variables here.
If needed, I can convert it to a helper function.

## Related Issues

closes #163

### CODE REVIEW

File: test/unit/ipv4_test.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+import (
+	"testing"
+
+	"github.com/go-playground/validator/v10"
+
+	"github.com/sivchari/govalid/test"
+)
+
+func TestIPV4Validation(t *testing.T) {
```
Comment (sivchari) [inline test/unit/ipv4_test.go]: Please use t.Parallel for top-level and sub-level tests.

**Review by sivchari — APPROVED** [review]
